packages/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -799,9 +824,16 @@ class DescopeWc extends BaseDescopeWc {
       this.rootElement.replaceChildren(clone);
 
       // we need to wait for all components to render before we can set its value
-      setTimeout(() =>
-        updateScreenFromScreenState(this.rootElement, screenState),
-      );
+      setTimeout(() => {
+        updateScreenFromScreenState(this.rootElement, screenState);
+
+        const eles = this.rootElement.querySelectorAll('descope-password');
+
+        // remove existing external inputs
+        document.querySelectorAll('[data-hidden-input="true"]').forEach(ele => ele.remove());
```

Comment:
Why do we need to do it?

Reply:
To clean up hidden elements after each screen